### ISSUE TITLE
Regression in 2.1: Bug with opacity of a Polygon2D

### ISSUE THREAD

**Issue by Rinocrosser** [godotengine/godot#6060]
**Operating system or device - Godot version:**
Windows 8.1 x64, Godot 2.1 RC2

**Issue description** (what happened, and what was expected):
I have a project, created in Godot 2.0.3 Stable. There are Polygon2D nodes, their opacity is not working in Godot 2.1 RC2. They are always 100% visible (making fog looks wierd). Tried to re-import textures, but it is not fixing problem. If I create new project and new scene in 2.1 RC2, opacity of the polygon is not working as well. Seems like something is changed in 2.1 making any polygon opcatiy working wrong.

**Steps to reproduce:**
In the description.

**Link to minimal example project** (optional but very welcome):

![godotbug](https://cloud.githubusercontent.com/assets/19660324/17461284/bad291f8-5c8f-11e6-9b7c-7ad9ddba64fe.jpg)
Fog must be barely visible and stars must blink, but opacity aint working, so the things looks wierd.

**Comment by akien-mga** [comment]
Can you create a minimal project in 2.0.4.1 that would trigger the bug in 2.1 RC2?

**Comment by Rinocrosser** [comment]
Sorry for the delay, I'll try to do it asap.

**Comment by Rinocrosser** [comment]
Okay, @akien-mga I was wrong about the core of the problem. I have a polygon to create fog shape, Polygon2D, it is filled with a texture. And I can't set the opacity of any polygon to anything but 100%.

**Comment by akien-mga** [comment]
Then it's likely a duplicate of #5151.

**Comment by Rinocrosser** [comment]
@akien-mga But in my case it working in 2.0.4 and 2.0.3, but not working in 2.1.

**Comment by akien-mga** [comment]
Yeah, I think #5151 is also about a regression, but it doesn't make it clear (and actually #5544 mentions a bug in 2.0.3). Let's reopen this one to investigate this issue as a regression, instead of as a "known bug that will be fixed in 3.0" as @reduz assumed in #5151.

**Comment by akien-mga** [comment]
If you could provide a test project that works in 2.0.4.1 and not in 2.1, that would be helpful to bisect the issue.

**Comment by Rinocrosser** [comment]
[TestBug6060.zip](https://github.com/godotengine/godot/files/410665/TestBug6060.zip)

@akien-mga Look at this one, the minimal set of one scene and one PNG file. Open with 2.0.4 - opacity works. Opening with 2.1 - it aint working at all.

**Comment by bojidar-bg** [comment]
I'm bisecting....

**Comment by bojidar-bg** [comment]
Ok, here is the result (not unexpected in any way):
````
0d20ceeb61be915fea81c22c174efb9253d96974 is the first bad commit
commit 0d20ceeb61be915fea81c22c174efb9253d96974
Author: Ovnuniarchos <pujalte.jorge@gmail.com>
Date:   Mon May 23 03:40:45 2016 +0200

    Polygon2D now exposes vertex colors.

:040000 040000 6175bde66dfbeba63cda2972425e91f5f24d5ff8 a22435a46c2723316dcdd5220fae4832fdaebff8 M      scene
````

**Comment by akien-mga** [comment]
CC @Ovnuniarchos.

**Comment by Ovnuniarchos** [comment]
The code for opacity could be implemented in the scene object, but that could be too hacky a way to solve the issue. The alternative is to drop support for vertex colors.

**Comment by akien-mga** [comment]
Is this still valid in 2.1.3 or the current 2.1 branch?

**Comment by eon-s** [comment]
Yes, the only way to change opacity on a Polygon2D is to change vertex color opacity or use a fragment shader (I guess vertex may work too).

**Comment by vnen** [comment]
Fixed by #9947.
